### PR TITLE
{KeyVault} Hotfix: Fix CloudSuffixNotSetException while using `az keyvault` in Azure Stack and Air-Gaped Cloud

### PR DESCRIPTION
**Pull request by fengzhou-msft**
Pick up the changes in #15298 for hotfix release.

**Description**  

**Testing Guide**  

**History Notes**  

[Component Name 1] BREAKING CHANGE: az command a: Make some customer-facing breaking change.  
[Component Name 2] az command b: Add some customer-facing feature.

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

### CODE REVIEW

**Review by Juliehzl — APPROVED** [review]

**Comment by yungezz** [comment]
will this cause problem when release back -> dev branch?

**Review by yungezz — APPROVED** [review]

File: src/azure-cli-core/azure/cli/core/cloud.py [hunk 1]
```diff
@@ -191,8 +191,8 @@ def _get_storage_sync_endpoint(cloud_name):
 
 def _get_synapse_analytics_endpoint(cloud_name):
     synapse_analytics_endpoint_mapper = {
-        'AzureCloud': 'dev.azuresynapse.net',
-        'AzureChinaCloud': 'dev.azuresynapse.azure.cn'
+        'AzureCloud': '.dev.azuresynapse.net',
```
Comment (yungezz) [inline src/azure-cli-core/azure/cli/core/cloud.py:194]: why change this? I think there's a param addDot=True.

File: src/azure-cli-core/azure/cli/core/cloud.py [hunk 2]
```diff
@@ -256,6 +256,11 @@ def _get_attestation_endpoint(cloud_name):
     return attestation_endpoint_mapper.get(cloud_name, None)
 
 
+def _get_mhsm_dns_suffix(cloud_name):
+    mhsm_dns_suffix_mapper = {c.name: c.suffixes.mhsm_dns for c in HARD_CODED_CLOUD_LIST}
+    return mhsm_dns_suffix_mapper.get(cloud_name, None)
```
Comment (yungezz) [inline src/azure-cli-core/azure/cli/core/cloud.py:261]: for air-gapped cloud, cloud_name isn't in HARD_CODED_CLOUD_LIST, then mhsm_dns_suffix_mapper will be None, line 261 will throw NoneType error?

File: src/azure-cli-core/azure/cli/core/cloud.py (hunk 1 above)
Comment (fengzhou-msft) [reply src/azure-cli-core/azure/cli/core/cloud.py:194]: addDot=True is for value from metadata url. This is the fallback value and is used directly.

Comment (fengzhou-msft) [reply src/azure-cli-core/azure/cli/core/cloud.py:194]: It's a mistake and caught by the test.

File: src/azure-cli-core/azure/cli/core/cloud.py (hunk 2 above)
Comment (fengzhou-msft) [reply src/azure-cli-core/azure/cli/core/cloud.py:261]: line 261 will return None as the value.

**Review by yungezz — APPROVED** [review]

**Pull request merged**